mutablesecurity/solutions/implementations/duplicati/code.py

```diff
@@ -0,0 +1,693 @@
+        )
+
+    IDENTIFIER = "local_backup"
+    DESCRIPTION = "Save files local"
+    ACT = local_backup
+
+
+class RestoreLocalBackup(BaseAction):
+    @staticmethod
+    @deploy
+    def restore_local_backup(source_file: str, backup_location: str) -> None:
+        command = DuplicatiHelper._make_local_backup(
+            source_file, backup_location, reverse=True
+        )
+        server.shell(
+            commands=[command],
+            name="Restore backup files on localhost.",
+        )
+
+    IDENTIFIER = "restore_local_backup"
+    DESCRIPTION = "Restore backup files on localhost"
+    ACT = restore_local_backup
+
+
+class SshBackup(BaseAction):
+    @staticmethod
+    @deploy
+    def ssh_backup(
+        source_file: str,
+        server_ip: str,
+        username: str,
+        password: str,
+        passphrase: str,
+    ) -> None:
+        command = DuplicatiHelper._make_ssh_backup(
+            source_file, server_ip, username, password, passphrase
+        )
+        server.shell(
+            commands=[command],
+            name="Save file over SSH",
+        )
+
+    IDENTIFIER = "ssh_backup"
+    DESCRIPTION = "Save files on remote computer via SSH"
+    ACT = ssh_backup
+
+
+class RestoreSshBackup(BaseAction):
+    @staticmethod
+    @deploy
+    def restore_ssh_backup(
+        source_file: str, server_ip: str, username: str, password: str
+    ) -> None:
+        command = DuplicatiHelper._make_ssh_backup(
+            source_file, server_ip, username, password, reverse=True
+        )
+        server.shell(
+            commands=[command],
+            name="Restore backup file over SSH",
+        )
+
+    IDENTIFIER = "restore_ssh_backup"
+    DESCRIPTION = "Restore files on localhost from remote computer over SSH"
+    ACT = restore_ssh_backup
+
+
+class GoogleDriveBackup(BaseAction):
+    @staticmethod
+    @deploy
+    def google_drive_backup(
+        source_file: str, backup_location: str, oauth_token: str
+    ) -> None:
+        command = DuplicatiHelper._make_googledrive_backup(
+            source_file, backup_location, oauth_token
+        )
+        server.shell(
+            commands=[command],
+            name="Save file to Google Drive",
+        )
+
+    IDENTIFIER = "google_drive_backup"
+    DESCRIPTION = "Save files to Google Drive"
+    ACT = google_drive_backup
+
+
+class RestoreGoogleDriveBackup(BaseAction):
+    @staticmethod
+    @deploy
+    def restore_google_drive_backup(
+        source_file: str, backup_location: str, oauth_token: str
+    ) -> None:
+        command = DuplicatiHelper._make_googledrive_backup(
+            source_file, backup_location, oauth_token, reverse=True
+        )
+        server.shell(
+            commands=[command],
+            name="Get backup file from Google Drive",
+        )
+
+    IDENTIFIER = "restore_google_drive_backup"
+    DESCRIPTION = "Get backup file from Google Drive"
+    ACT = restore_google_drive_backup
+
+
+# info
```

Comment:
Maybe a forgot comment? The comment dividing the classes is already present below.